src/app/pages/wallets/wallet-details/pages-wallet-details.component.ts

```diff
@@ -1,28 +1,25 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
-import { FormControl } from '@angular/forms';
+import { Component, OnInit, ViewChild } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
-import { Subscription } from 'rxjs';
+import {
+  TransactionTypeMatSelectComponent,
+  WalletSelectionValue,
+} from 'src/app/common/components/mat-controls/transaction-type-mat-select/transaction-type-mat-select.component';
 import { TDataState } from 'src/app/common/http/common.http.types';
 import { WalletTransactionType } from 'src/app/domains/wallets/domains.wallets.types';
 import { WalletsManagementItem } from '../management/pages-wallets-wallets-management-item.model';
 import { PagesWalletsManagementEditorService } from '../management/wallet-editor/pages-wallets-management-editor.service';
 import { IWalletModalData } from '../management/wallet-editor/pages-wallets-management-editor.types';
 import { WalletsDetailsTransaction } from './pages-wallet-details-item.model';
 import { PagesWalletDetailsService } from './pages-wallet-details.service';
-
-type WalletSelectionValue = WalletTransactionType | '';
+import { PagesWalletTransactionEditorService } from './transaction-editor/pages-wallet-transaction-editor.service';
 
 @Component({
   selector: 'app-pages-wallet-details',
   templateUrl: './pages-wallet-details.component.html',
   styleUrls: [ './pages-wallet-details.component.scss' ],
 })
-export class PagesWalletDetailsComponent implements OnInit, OnDestroy {
-  public selectTransactionsTypes: Record<string, WalletSelectionValue> = {
-    'All transactions': '',
-    'Incomes': WalletTransactionType.Incomes,
-    'Expenses': WalletTransactionType.Expenses,
-  };
+export class PagesWalletDetailsComponent implements OnInit {
+  @ViewChild('transactionTypeMat') public transactionTypeMat!: TransactionTypeMatSelectComponent;
```

Comment:
Again - a bit of duplication is better than too clever abstraction - for example you now have an issue with labels, on wallet details page for filtering the table it should have options:
- All
- Incomes
- Expenses
and for editor it should be:
- Income
- Expense
and now you have to handle that in this component instead simple logic in two different components. And later you can have even more scenarios like this + what about validating in editor that this field was selected? Will validation work as easy as in case of using simple `FormControl` and `Validators.required`?